src/track/keyutils.cpp

```diff
@@ -17,7 +17,7 @@ using mixxx::track::io::key::ChromaticKey_IsValid;
 static const QString s_openKeyPattern("^\\s*(1[0-2]|[1-9])([dm])\\s*$");
 
 // Lancelot notation, the numbers 1-12 followed by a (minor) or b (major).
-static const QString s_lancelotKeyPattern("^\\s*(1[0-2]|[1-9])([ab])\\s*$");
+static const QString s_lancelotKeyPattern("^\\s*(1[0-2]|0?[1-9])([ab])\\s*$");
```

Comment:
I struggle to understand what that `0?` is doing there? I guess accounting for zero-padded numbers? Wouldn't it make more sense then to be even more promiscous and allow an arbitrary number of preceding 0's instead of just when padded 2 chars?
```suggestion
static const QString s_lancelotKeyPattern("^\\s*0*(1[0-2]|[1-9])([ab])\\s*$");
```

Reply:
Yes, this is for the compatibility with the Rapid Evolution "Key Code" notation using always two digits like "01A". This was the easiest way to make it compatible to Mixxx's Lancelot format which does not have the zero padding.

I think I need to add a comment that describes that.  

I do not know any formats with more leading zeros so I do not see a point to allow that. Do you?

Reply:
I don't know any either, still just padding a single 0 instead of arbitrary many seems like a missed opportunity, doesn't it?